cpp/src/io/json/reader_impl.cu

```diff
@@ -345,15 +337,19 @@ void reader::impl::set_record_starts(rmm::cuda_stream_view stream)
         filtered_count--;
       }
     }
-
-    rec_starts_ = h_rec_starts;
-    thrust::sort(rmm::exec_policy(stream), rec_starts_.begin(), rec_starts_.end());
+    CUDA_TRY(cudaMemcpyAsync(rec_starts.data(),
```

Comment:
This copy and the subsequent sort may not complete before the function returns and h_rec_starts is destroyed.  So probably need to stream.synchronize() after the sort.